sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,304 @@
+
+    @click.command()
+    @click.option(
+        "-s",
+        "--schedule",
+        default="",
+        help=(
+            "The cron schedule expression to define when to run tempest"
+            " periodic checks. When the value is empty (default),"
+            " period checks will be disabled."
+        ),
+    )
+    def configure_validation(self, schedule: str = "") -> None:
+        """Configure validation plugin."""
+        jhelper = JujuHelper(self.client, self.snap.paths.user_data)
+        with console.status("Configuring validation plugin ..."):
+            valid, message = validate_schedule(schedule)
+            if not valid:
+                raise click.ClickException(message)
+
+            app = "tempest"
+            model = OPENSTACK_MODEL
+            unit = run_sync(jhelper.get_leader_unit(app, model))
+            if not unit:
+                message = f"Unable to get {app} leader"
+                raise click.ClickException(message)
+
+            run_sync(
+                jhelper.set_application_config(
+                    model, app, config={"schedule": schedule}
+                )
+            )
+            console.print(message)
+
+    @click.command()
+    @click.option(
+        "-s",
+        "--smoke",
+        is_flag=True,
+        default=False,
+        help="Run the smoke tests only. Equivalent to --regex=smoke.",
+    )
+    @click.option(
+        "-r",
+        "--regex",
+        default="",
+        help=(
+            "A list of regexes, whitespace separated, used to select tests from"
+            " the list."
+        ),
+    )
+    @click.option(
+        "-e",
+        "--exclude-regex",
+        default="",
+        help="A single regex to exclude tests.",
+    )
+    @click.option(
+        "-t",
+        "--serial",
+        is_flag=True,
+        default=False,
+        help="Run tests serially. By default, tests run in parallel.",
+    )
+    @click.option(
+        "--test-list",
+        default="",
+        help=(
+            "Use a predefined test list. See `sunbeam validation-lists`"
+            " for available test lists."
+        )
+    )
+    def run_validate_action(
+        self,
+        smoke: bool = False,
+        regex: str = "",
+        exclude_regex: str = "",
+        serial: bool = False,
+        test_list: str = "",
+    ) -> None:
+        """Validate the sunbeam installation."""
+        action_name = "validate"
+        action_params = {
+            "regex": "smoke" if smoke else regex,
+            "exclude-regex": exclude_regex,
+            "serial": serial,
+            "test-list": test_list,
+        }
+        progress_message = "Running tempest to validate the sunbeam deployment ..."
+        self._run_action(
+            action_name,
+            action_params=action_params,
+            progress_message=progress_message,
+            print_stdout=True,
+        )
+
+    @click.command()
+    def run_get_lists_action(self) -> None:
```

Comment:
I've been thinking about future extensibility when we want to show the content of a test lists. Maybe it can be nested under this command with something like `sunbeam validation-lists --show readonly-quick`. Anyway, this is for a future iteration and I just want to put a note here.